### PR TITLE
Add --auto option to automatically pick files to format

### PR DESCRIPTION
**Pull request by V-FEXrt**
This change introduces the `--auto`  flag which uses the built in wake mechanism to discover which files should be formatted.

This means that 

`wake-format --auto --dry-run` is sufficient to ensure all wake files discovered by wake will be formatted and `wake-format --auto --in-place` will format all said files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,7 +56,7 @@ install:	all
 # It assumes clang is available on the PATH and will fail otherwise
 formatAll:
 	@clang-format -i --style=file $(shell ./scripts/which_clang_files all)
-	@bin/wake-format.native-cpp14-release -i $(shell ./scripts/which_wake_files all)
```
Comment (JakeSiFive) [inline Makefile:59]: Can we delete that script now?

Comment (JakeSiFive) [reply Makefile:59]: Also was this script missing build.wake and tests.wake?

File: build.wake [hunk 2]
```diff
@@ -187,35 +187,41 @@ def tarball Unit =
     def wakeSourcesString =
         wakePaths
         | map (_.getPathName.format)
-        | foldr (_, ",\n  ", _) Nil
+        | foldr (_, ",\n    ", _) Nil
```
Comment (JakeSiFive) [inline build.wake:190]: Why is this space being added?

Comment (V-FEXrt) [reply build.wake:190]: See the other comment; tl;dr it ensures `manifest.wake` is formatted

Comment (JakeSiFive) [reply build.wake:190]: Ah ok gotcha

File: Makefile (hunk 1 above)
Comment (JakeSiFive) [reply Makefile:59]: Ah gotcha

**Review by JakeSiFive — APPROVED** [review]

**Pull request merged**